### DISCUSSION TITLE
Support Windows and Linux operating systems

### DISCUSSION THREAD

**Discussion by mattstauffer** [tighten/lambo#160] [Ideas]
### Detail to resolve

**Linux**

> What Linux distros should we support?

> The community Valet for Linux port is flaky, do we support it or do something else?

**Windows**

> Do we even support Windows?

**A**: Maybe we just advise the use of Laragon. 
**A**: Maybe support is indirect through WSL and Lambo's supported Linux distros.

> What Windows versions should we support?

**A**: Only those that have WSL/WSL2? 

> How do we replace valet functionality?

**A** We don't (at least for launch of official compatibility)

**Comment by jonsugar** [comment]
@mattstauffer Is this satisfied by #167 ? In other words, Lambos position would be it provides Sail support but does not aim to provide full parity of functionality with non-Mac operating systems.

**Reply by jonsugar** [reply]
mattstauffer said:
>@jonsugar I think the main thing I would want to make sure is that the syntax we use for all our commands works fine on windows >and linux. If so, then yes!

**Comment by benholmen** [comment]
Hi it's me, your Windows WSL2 test subject! 👋 

I tried running `lambo new hello-world` expecting Lambo to just skip the Valet steps since it wasn't available. My expectation was that it'd do the following:

- `laravel new $PROJECTNAME`
- Initialize a git repo, add all of the files, and, after some changes below, make a commit with the text "Initial commit."
- Replace the `.env` (and `.env.example`) database credentials with ~~the default macOS MySQL credentials: database of `$PROJECTNAME`, user `root`, and empty password~~ sqlite configuration; run `touch database/database.sqlite`
- Replace the `.env` (and `.env.example`) `APP_URL` with ~~`$PROJECTNAME.$YOURVALETTLD`~~ `$PROJECTNAME.local` or a config file TLD
- Generate an app key
- Open the project in your favorite editor
- Open ~~`$PROJECTNAME.$YOURVALETTLD`~~ `$PROJECTNAME.local` in your browser

In general I think `$YOURVALETTLD` could default to:
1. Your Valet TLD if Valet is available 👍 
1. A config value
1. A parameter value
1. A sensible default like `.local` or `.dev` (probably whatever Valet uses by default cause y'all like Valet)

My naive opinion is that Lambo should use Valet (if available) or Sail (if available) or just do what it can, like I outlined above.
